examples/intro/inducing_path.py

```diff
@@ -0,0 +1,116 @@
+"""
+======================================================
+An introduction to Inducing Paths and how to find them
+======================================================
+
+A path p is called an ``inducing path`` relateve to <L,S>
+on an ancestral graph, where every non-endpoint vertex on p
+is either in L or a collider, and every collider on p is an ancestor
+of either X , Y or a member of S.
+
+
+In other words, it is a path between two nodes that cannot be
+d-seperated, making it active regardless of what variables
+we condition on.
+
+More details on inducing paths can be found at :footcite:Zhang2008.
+
+References
+----------
+.. footbibliography::
+"""
+
+import pywhy_graphs
+from pywhy_graphs import ADMG
+from pywhy_graphs.viz import draw
+
+# %%
+# Construct an example graph
+# ---------------------------
+# To illustrate the workings of the inducing path algorithm, we will
+# construct the causal graph from figure 2 of :footcite:`Colombo2012`.
+
+
+G = ADMG()
+G.add_edge("X4", "X1", G.directed_edge_name)
+G.add_edge("X2", "X5", G.directed_edge_name)
+G.add_edge("X2", "X6", G.directed_edge_name)
+G.add_edge("X4", "X6", G.directed_edge_name)
+G.add_edge("X3", "X6", G.directed_edge_name)
+G.add_edge("X3", "X4", G.directed_edge_name)
+G.add_edge("X3", "X2", G.directed_edge_name)
+G.add_edge("X5", "X6", G.directed_edge_name)
+G.add_edge("L2", "X4", G.directed_edge_name)
+G.add_edge("L2", "X5", G.directed_edge_name)
+G.add_edge("L1", "X1", G.directed_edge_name)
+G.add_edge("L1", "X2", G.directed_edge_name)
+
+
+# this is the Figure 2(a) in the paper as we see.
+dot_graph = draw(G)
+dot_graph.render(outfile="pag.png", view=True)
+
+
+# %%
+# Adjacent nodes trivially have an inducing path
+# -----------------------------------------------
+# By definition, all adjacent nodes have a trivial inducing path between them,
+# that path only consists of one edge, which is the edge between those two nodes.
+
+L = {}
+S = {}
+
+# All such tests will return True.
+print(pywhy_graphs.inducing_path(G, "X1", "X4", L, S))
+print(pywhy_graphs.inducing_path(G, "X3", "X2", L, S))
+
+# %%
+# Inducing paths between non-adjacent nodes
+# ------------------------------------------
+# Given the definition of an inducing path, we need to satisfy all
+# requirements for the function to return True. Adding the latent
+# variables to L is not enough for the pair [X1,X5]
```

Comment:
```suggestion
# variables to L is not enough for the pair [X1,X5]. As we see in Figure 2(c) in :footcite:`Colombo2012`,
# (X1, X5) are not adjacent in the final skeleton of the equivalence class, which makes sense because a
# MAG is an equivalence class of a DAG and contains an edge among two nodes if i) the two nodes are
# adjacent in the DAG, or ii) the two nodes have a primitive inducing path between them. Since there
# is no adjacency among (X1, X5) in the final skeleton, there is no primitive inducing path between them.
```